lib/metadata/wrapper.js

```diff
@@ -147,6 +147,31 @@ const metadata = {
     switch: newClient => {
         client = newClient;
     },
+
+    checkHealth: (log, cb) => {
+        if (!client.healthcheck) {
+            const defResp = { metadata: { code: 200, message: 'OK' } };
+            return cb(null, defResp);
+        }
+        return client.healthcheck(log, (err, result) => {
+            const respBody = {};
+            if (err) {
+                log.error(`error from ${implName}`, { error: err });
+                respBody[implName] = {
+                    error: err,
+                };
+                // respBody also returned so error is written to response
+                return cb(err, respBody);
+            }
+            const parseResult = JSON.parse(result);
```

Comment:
the json.parse should just be happening in bucketclient/backend.js instead of twice.  Actually, if we're only going to do the healthcheck for the bucketclient backend (rather than bucketfile and mem), this function should check whether the client supports the function and return as you have above and if it does support the function, it must be bucketclient which is filtered through bucketclient/backend.js so can just pass through the object from there.